junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
```

Comment:
Even though this file is called `Assertions.kt` what really happens is this creates a class called `AssertionsKt`. Thus preventing the naming conflict.
If the team would like to be more specific about the name of the class created we could add something like `@file:JvmName("AssertionsForKotlin")` to the top of the file.

This will require some monkeying with spotless due to this issue:
https://github.com/diffplug/spotless/issues/136

Does the team mind that the kotlin compiler just generates a class and uses its own naming convention? If not then I don't need to worry about the spotless bug.

Reply:
Is this naming convention likely to change in the future?

Reply:
Nope, it shouldn't.